### ISSUE TITLE
Question about gradient calculation.

### ISSUE THREAD

**Issue by NAM-hj** [noahchalifour/rnnt-speech-recognition#19]
Thank you for your code!
It's really helpful.

I have a question about the loss function.
I know the loss of the rnnt is this.
> loss = -ln P( y* | x)
>
>

At the utils.loss.get_loss_fn()
I can see the forward propagation of rnnt_loss.
I think **get_loss_fn()** of warp-transducer/tensorflow_binding/warprnnt_tensorflow/__init__.py is same with **utils.loss.get_loss_fn()** and **the function** called by 
`from warprnnt_tensorflow import rnnt_loss`

So I understand "get_loss_fn()" only calculate forward propagation.

Then, where is the backward propagation to update weights?
I only can see these codes. (run_common_voice.py)
```
# This is the code of run_common_voice.py (run_training() -> run_training() -> step_fn())

with tf.GradientTape() as tape:
    ...
    loss = loss_fn (labels, outputs, spec_lengths, label_lengths)
    ...
gradients = tape.gradient(loss, model.trainable_variables)
optimizer.apply_gradients(zip(gradients, model.trainable_variables))
```
I don't think this code call **the gradient calculation code of warp-transducer library**.

**How can it calculate the gradients??**

Thank you.

**Comment by noahchalifour** [comment]
@nambee The `GradientTape` automatically tracks variables and then the `tape.gradient` function computes the gradients given the loss and the variables to compute gradients for. Take a look at the Tensorflow documentation (https://www.tensorflow.org/api_docs/python/tf/GradientTape)
